### DISCUSSION TITLE
Matlab: question about the aircraft behaviour

### DISCUSSION THREAD

**Discussion by MariaALJ** [JSBSim-Team/jsbsim#804] [Q&A]
Hello,

I may have not understood something, but why phi is increasing (a lot) when i only used the elevator command with the throttle? I'm excpecting the aircaft only to have a longitudinal flight with these inputs. Is there any explanation for this behaviour? Same question for psi that is also increasing...
maybe @noah-curran you can help me on this matter?
Thank you very much in advance!

![image](https://user-images.githubusercontent.com/116007762/210544986-5a7eb753-9f08-4766-ab83-e4ea2ee530c0.png)

**Comment by agodemar** [comment]
Quick answer: because it has 6 degrees of freedom and the external forces and moments may be nonzero with zero sideslip.

**Comment by seanmcleod** [comment]
> ...when i only used the elevator command with the throttle? I'm excpecting the aircaft only to have a longitudinal flight with these inputs.

You're assuming that the aircraft was trimmed before you input your elevator and throttle changes.

**Reply by MariaALJ** [reply]
Hello, thank you @agodemar and @seanmcleod for your relpies. I understand much more aspects after our discussion #774. But @agodemar can you tell me please how to practically set the initial trimmed state?

I allow myself to quote @seanmcleod 's comment from #774 that I need your help with : 

> So the Simulink input values I assume will immediately overwrite the trim control positions 1 time step later which means the aircraft will no longer be in trim.

> @agodemar @noah-curran assuming you're using JSBSim's trim routine, as opposed to using a Matlab/Simulink trim how do you get the Simulink input values to be set to the JSBSim calculated trim control values so that the aircraft stays in trim? I glanced at the JSBSim S-Function source code and didn't see any particular code that might do this.

**Reply by agodemar** [reply]
@MariaALJ at the moment I am too busy to provide a complete answer. 

You did not tell about your background, but I think you could consider having a look at this YouTube playlist: [Flight Mechanics by Christopher Lum](https://www.youtube.com/playlist?list=PLxdnSsBqCrrEx3A6W94sQGClk6Q4YCg-h). In particular, about trim you have this video: [Trimming a Model of a Dynamic System Using Numerical Optimization](https://youtu.be/YzZI1V2mJw8). It uses Matlab and Simulink in his examples.

If we want to have JSBSim do the trim and fetch the trim results to the Simulink S-Function, I guess some modifications should be made to the C++ code. An alternative could be: not using a JSBSim script and use Simulink - in particular, its [Control Design toolbox](https://www.mathworks.com/products/simcontrol.html) - to do the trim.

**Reply by MariaALJ** [reply]
Thank you for your response. 
Actually I specialized in turbomachinery, aeroengine performance and combustion so my knowledge is basically about the engine and much less about the aircraft. But anyway, thank you for these references and advice, it will sure help figure it out!
